### PR TITLE
UniRef: add urlReferenceMap to RelationalView

### PR DESCRIPTION
**Pull request by Beanow**
`RelationalView` provides easy access to `ReferentEntities`, which we can use for reference detection. The map produced by `urlReferenceMap` can be used easily by a `MappedReferenceDetector`.

As discussed in #1532 we'll use `RelationalView` for this before deprecating it.

Test plan: `yarn unit`

### CODE REVIEW

**Comment by decentralion** [comment]
One more thing: in the RelationalView test file, we have individual examples of each entity type, e.g. here:

https://github.com/sourcecred/sourcecred/blob/adec1b0b5d40ca364873b0b725015b82ae6e515d/src/plugins/github/relationalView.test.js#L116

rather than having a snapshot test for the entire urlReferenceMap, lets just check that these entities are present in the map as expected. cuts down on the snapshot size (huge snapshots are an antipattern, as a reviewer my eyes glaze over). it feels like we shouldn't really need the snapshot at all.

**Comment by Beanow** [comment]
> One more thing: in the RelationalView test file, we have individual examples of each entity type, e.g. here:

Good point. I'm not sure how to add a smaller test repository, without using "private" methods on RelationalView (which is our test subject, so I'm hesitant to do that), other than hand-crafting a new repository json.

File: src/plugins/github/relationalView.js [hunk 1]
```diff
@@ -515,6 +515,30 @@ export class RelationalView {
     }
   }
 
+  /**
+   * Creates a Map<string, N.RawAddress> for each ReferentEntity in this view.
+   * Note: duplicates are accepted within one view. However for any URL, the
+   * corresponding N.RawAddress should be the same, or we'll throw an error.
+   */
+  urlReferenceMap() {
```
[the remark this answers is not in the document]
Comment (Beanow) [reply src/plugins/github/relationalView.js]: Done, switched the aliases over to `<URL, NodeAddressT>` as well to be more in line with the rest of reference detection.

File: src/plugins/github/relationalView.test.js [hunk 2]
```diff
@@ -390,6 +391,49 @@ describe("plugins/github/relationalView", () => {
...
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"REPO\\",\\"sourcecred-test\\",\\"example-github\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"ISSUE\\",\\"sourcecred-test\\",\\"example-github\\",\\"1\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"PULL\\",\\"sourcecred-test\\",\\"example-github\\",\\"3\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"REVIEW\\",\\"sourcecred-test\\",\\"example-github\\",\\"5\\",\\"100313899\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"ISSUE\\",\\"sourcecred-test\\",\\"example-github\\",\\"2\\",\\"373768703\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"PULL\\",\\"sourcecred-test\\",\\"example-github\\",\\"3\\",\\"369162222\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMENT\\",\\"REVIEW\\",\\"sourcecred-test\\",\\"example-github\\",\\"5\\",\\"100313899\\",\\"171460198\\"]",
+          "NodeAddress[\\"sourcecred\\",\\"github\\",\\"COMMIT\\",\\"MDY6Q29tbWl0MTIzMjU1MDA2OjBhMjIzMzQ2YjRlNmRlYzAxMjdiMWU2YWE4OTJjNGVlMDQyNGI2NmE=\\"]",
+        ]
+      `);
```
Comment (Beanow) [inline src/plugins/github/relationalView.test.js]: @decentralion How is this as an alternative to the mega snapshot?

Comment (decentralion) [reply src/plugins/github/relationalView.test.js]: This is basically what I had in mind. However, to make it easier to read, I recommend creating an array of {url, address} objects, and snapshotting that array. that way the reader can directly correlate the address to the url without needing to mentally track an index in both arrays.

Comment (Beanow) [reply src/plugins/github/relationalView.test.js]: I tried a few options.

```js
type MapBased = Map<string, ?string>;
type ObjectBased = {url: string, address: ?string}[];
type TupleBased = [string, ?string][];
```

And found the tuple version most easy to read in the snapshot (how prettier stores it) and test error message context. All of them were either a lot larger than the inline snapshot, so I've moved that back to the snapshot file.

**Comment by Beanow** [comment]
Requesting review for the test readability change anyway, as I don't see a need to rush merge this :]

**Pull request merged**